subprojects/problems/src/main/java/org/gradle/api/problems/interfaces/ProblemLocation.java

```diff
@@ -14,17 +14,21 @@
  * limitations under the License.
  */
 
-package org.gradle.problems
+package org.gradle.api.problems.interfaces;
 
-import groovy.transform.Canonical
-import groovy.transform.CompileStatic
+import org.gradle.api.Incubating;
 
-@CompileStatic
-@Canonical
-class TestContext {
-    final String value
+import javax.annotation.Nullable;
 
-    String toString() {
-        value
-    }
+/**
+ * Problem location.
+ *
+ * @since 8.3
+ */
+@Incubating
+public interface ProblemLocation {
+    String getPath();
```

Comment:
💭 Absolute path only, right?  Should the `org.gradle.problems.Location` type be reused for this?

Reply:
The situation has evolved since then, with all the location types added.
The `FileLocation` type will be used, which is based on this interface.